Alignment/HIPAlignmentAlgorithm/scripts/hippyaddtobaddatafiles.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+import argparse, contextlib, os, re, shutil, subprocess, tempfile, time
+
+if __name__ == "__main__":
+  def abspath(path):
+    if not os.path.exists(path): raise ValueError(path+" does not exist")
+    return os.path.abspath(path)
+
+  p = argparse.ArgumentParser()
+  p.add_argument("cfgfile", type=abspath)
+  p.add_argument("baddatafileslist", nargs="?", default=None)
+  args = p.parse_args()
+
+def runcfg(cfgfile, badfilelist):
+  try:
+    subprocess.check_output(["cmsRun", cfgfile], stderr=subprocess.STDOUT)
+  except subprocess.CalledProcessError as e:
+    if "FallbackFileOpenError" in e.output:
+      output = e.output.split("An exception of category 'FallbackFileOpenError' occurred while")[1]
+      filename = re.search("Failed to open the file '[^']*(/store/.*[.]root)", output).group(1)
+      with OneAtATime(badfilelist+".tmp", 2) as f:
+        with open(badfilelist) as f:
+          contents = set(f.read().split())
+        if filename in contents:
+          raise RuntimeError(filename+"\nis already in\n"+badfilelist+"\n\nExiting to avoid an infinite loop.  Maybe you have this running on the same cfg file multiple times?")
+        contents.add(filename)
+        contents = sorted(contents)
+        with open(badfilelist, "w") as f:
+          f.write("\n".join(contents)+"\n")
+        print "found and added a bad file:\n"+filename
```

Comment:
@hroskes could you please make this python3 compliant adding parentheses to the print statements and 
from __future__ import print_function
at the beginning?